codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -176,18 +175,18 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
             // for this stream.
             Http2RemoteFlowController flowController = flowController();
             if (!endOfStream || !flowController.hasFlowControlled(stream)) {
-                ChannelFuture future = frameWriter.writeHeaders(ctx, streamId, headers, streamDependency, weight,
-                                                                exclusive, padding, endOfStream, promise);
                 if (endOfStream) {
                     final Http2Stream finalStream = stream;
-                    future.addListener(new ChannelFutureListener() {
+                    final ChannelFutureListener closeStreamLocalListener = new ChannelFutureListener() {
```

Comment:
@buchgr was there any reason why we needed to assign this and could not just add the instance directly like before ?